[docs] Create modchat spec (#78)
cscareerhub/megabot: specs/modchat.md

> # Modchat

Reviewer: rename file to `modchat.spec.md`

> Used to anonymously private message bot and information to be relayed to staff team

Reviewer: ```suggestion
Allows users to anonymously message or report something to the whole staff team by private messaging the bot.
```

1. Responds to `++mc`.
2. When private messaged with `++mc` as starting it will forward message to a designated channel on server.

> 3. Add `-a` flag (so `++mc -a`) to have the message anonymous to staff.

Reviewer: ```suggestion
3. `++mc a` anonymizes the message sent to staff.
```

> 4. Allow mods to designate target channel for messages with `++mc designate` in the target channel.

Reviewer: ```suggestion
4. Allow mods to designate target channel for messages by typing `++mc designate` in the target channel.
```
